### PR TITLE
[Solved #27] Implement routes

### PR DESCRIPTION
**Pull request by supunTE**
Solved #27

Implemented routing

This PR is built on `supunTE:feature/25-base-layout` (PR - #26)

### CODE REVIEW

File: apps/client/src/pages/leaderboard/Leaderboard.page.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+export function Leaderboard(): ReactElement {
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    // after 2 seconds redirect to room
+    setTimeout(() => {
+      navigate('../room');
+    }, 2000);
+  }, []);
```
Comment (dalanad) [inline apps/client/src/pages/leaderboard/Leaderboard.page.tsx:16]: Shall we return a callback to clear the timeout

File: apps/client/src/pages/home/Home.page.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    } else {
+      const tournamentId = getTournamentId();
+      navigate(`/${tournamentId}/room`);
+    }
+  };
+
+  return (
+    <div>
+      <h1>Home</h1>
+      {id && <p>(id: {id})</p>}
```
Comment (dalanad) [inline apps/client/src/pages/home/Home.page.tsx]: lets use a ternary operation with a null value instead of `&&`

File: apps/client/src/main.tsx [hunk 3]
```diff
@@ -12,7 +12,20 @@ const root = ReactDOM.createRoot(
...
+          <Route path='/' element={<Layout />}>
+            <Route index element={<Home />} />
+            <Route path=':id'>
+              <Route index element={<Home />} />
+              <Route path='room' element={<Room />} />
+              <Route path='race' element={<Race />} />
+              <Route path='leaderboard' element={<Leaderboard />} />
+            </Route>
+          </Route>
+          <Route path='*' element={<Navigate to='/' />} />
```
Comment (dalanad) [inline apps/client/src/main.tsx]: Better to add a 404 page from start

**Review by dalanad — CHANGES_REQUESTED** [review]

File: apps/client/src/pages/leaderboard/Leaderboard.page.tsx (hunk 1 above)
Comment (supunTE) [reply apps/client/src/pages/leaderboard/Leaderboard.page.tsx:16]: added cleanup function 063125405e6f093999cfaf77d5e479406ddea7a9

File: apps/client/src/pages/home/Home.page.tsx (hunk 2 above)
Comment (supunTE) [reply apps/client/src/pages/home/Home.page.tsx]: changed a0b11062df50938955e49355176719383d577e6a

File: apps/client/src/main.tsx (hunk 3 above)
Comment (supunTE) [reply apps/client/src/main.tsx]: Added 0552de4faf3c4060bf3af438fa8cb0eb6467254b

**Review by dalanad — APPROVED** [review]

**Review by pasinducw — APPROVED** [review]

**Pull request merged**